### DISCUSSION TITLE
Wrong IP showing - Docker

### DISCUSSION THREAD

**Discussion by Deck8** [AzuraCast/AzuraCast#7397] [General]
Using azuracast docker and have bound ports to a specific IP. The SFTP information however shows incorrect IP in the "You may need to connect directly to your IP address:" is there a way of changing this to reflect correct IP?

I use CF proxying for all the web interface/streams meaning IP/real hostname is needed for connecting to SFTP so would be nice to have correct option displayed or an option in system settings to change what hostname/ip is displayed for SFTP information.

EDIT: Just to add same applies to streaming and DJ information as well.

**Comment by BusterNeece** [comment]
@Deck8 We fetch that IP by pinging our central updates server, and having it report back the IP that it sees from your installation. (We don't store the IP, it's just an echo-back service as it's all but impossible for us to determine the outward-facing IP from within the Docker container)

I'm curious why it would be showing the wrong IP in your case...unless you're specifically expecting it to show an internal LAN IP?

**Reply by Deck8** [reply]
@BusterNeece the server has multiple IP addresses so the IP address shown is the main server IP rather than the one docker ports are bound to. All generic outbound traffic would use the main IP as you can see below 144.217.x.x is IP address reported in azuracast (and if I curl an address) but 198.100.x.x is actual IP used to access docker

```
# curl icanhazip.com
144.217.*.*

# ip a
2: eno3: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP group default qlen 1000
    inet 198.100.*.*/32 scope global eno3
       valid_lft forever preferred_lft forever
    inet 144.217.*.*/24 brd 144.217.*.* scope global dynamic eno3
       valid_lft 45398sec preferred_lft 45398sec
```

In this case is would it be possible to add an (advaced) option to allow this displayed IP to be overridden manually? So an admin can specify what correct external IP is if auto-detection is incorrect.

**Comment by wolfstep1** [comment]
I am having the same issue.
I have 5 static IPs, my azurecast is dedicated to one of the IPs.
My PFSense is set to nat all traffic for the azurecast domain to the local ip.
When I go into the Azurecast, it shows the first of my five numbers, not the second like it should.
If Azurecast is "pinging" your server, using your statement above, should it not be coming back to the proper IP?
I am trying to get this server ready for live traffic, but it showing the wrong IP will surely create grief for my users.
It does however show the correct domain name.
